py/legacypipe/oneblob.py

```diff
@@ -143,11 +184,15 @@ def __init__(self, name, blobwcs, blobmask, timargs, srcs, bands,
         # else:
         #     self.optargs.update(dchisq = 0.1)
 
-        from legacypipe.constrained_optimizer import ConstrainedOptimizer
-        self.trargs.update(optimizer=ConstrainedOptimizer())
+        #from tractor.constrained_optimizer import ConstrainedOptimizer
+        #self.trargs.update(optimizer=ConstrainedOptimizer())
+        from tractor.dense_optimizer import ConstrainedDenseOptimizer
+        self.trargs.update(optimizer=ConstrainedDenseOptimizer())
+
         self.optargs.update(dchisq = 0.1)
 
-    def run(self, B):
+    def run(self, B, reoptimize=False, iterative_detection=True,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [OneBlob.run is too complex (19)](https://app.codacy.com/app/dstndstn/legacypipe/pullRequest?prid=4730106)